src/textual/message.py

```diff
@@ -73,8 +73,16 @@ def __init_subclass__(
             cls.no_dispatch = no_dispatch
         if namespace is not None:
             cls.namespace = namespace
-        name = camel_to_snake(cls.__name__)
-        cls.handler_name = f"on_{namespace}_{name}" if namespace else f"on_{name}"
+            name = f"{namespace}_{camel_to_snake(cls.__name__)}"
+        else:
+            # a class defined inside of a function will have a qualified name like func.<locals>.Class,
+            # so make sure we only use the actual class name(s)
+            qualname = cls.__qualname__.rsplit("<locals>.", 1)[-1]
```

Comment:
This seem brittle. Is it guaranteed that CPython and all other Python's will have a qualname formatted like this?

Reply:
It's explicitly defined in the PEP that introduced `__qualname__`: https://peps.python.org/pep-3155/#proposal